frontend/src/providers/UserProvider.jsx

```diff
@@ -0,0 +1,19 @@
+import React, { createContext,useState,useEffect } from "react";
+import { auth } from "../firebase";
+
+export const UserContext = createContext({ user: null });
+
+const UserProvider = (props) => {
+  const [user, setUser] = useState();
+  useEffect(() => {
+    auth.onAuthStateChanged((userAuth) => {
+      setUser({ user: userAuth });
+    });
```

Comment:
Two things:
1) the `setUser` function always updates `user`, that's its only purpose. So on line 10 you can simply write `setUser(userAuth)` and it will automatically set `user` to that value.
2) Remember it's best to add a dependency array at the end of `useEffect` so it doesn't get called constantly. How often do you want this to update?